src/graph/executor/query/FilterExecutor.cpp

```diff
@@ -65,7 +65,9 @@ StatusOr<DataSet> FilterExecutor::handleJob(size_t begin, size_t end, Iterator *
   for (; iter->valid() && begin++ < end; iter->next()) {
     auto val = condition->eval(ctx(iter));
     if (val.isBadNull() || (!val.empty() && !val.isImplicitBool() && !val.isNull())) {
-      return Status::Error("Wrong type result, the type should be NULL, EMPTY, BOOL");
+      return Status::Error("Failed to evaluate condition: %s. %s",
+                           condition->toString().c_str(),
+                           "For boolean conditions, please write <condition> == <true/false>.");
```

Comment:
prefer the error message: `it is not a boolean condition.` sometimes the condition could be `X.xx IS NULL`, which is not the format such as `<condition> == <true/false>`

Reply:
Fixed. Need to explicitly recommend a way to write conditions here to avoid running into errors.